### PR TITLE
Added Resource Classes and Competition Classes

### PR DESCRIPTION
**Pull request by satkab**
I added all of the resource classes and competition class to the 6220_2021 folder and I cleaned up the code from last year and added new methods for this year's robot.

### CODE REVIEW

File: Team6220_2021/src/main/java/org/firstinspires/ftc/team6220_2021/MasterAutonomous.java [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+
+        while (!distanceReached && opModeIsActive()) {
+            // This calculates the angle deviation
+            angleDeviation = imu.getAngularOrientation(AxesReference.INTRINSIC, AxesOrder.ZYX, AngleUnit.DEGREES).firstAngle - startAngle;
+
+            // This adds a value to the PID loop so it can update
+            distanceLeft = targetDistance - position;
+            translationPID.roll(distanceLeft);
+
+            // We drive the mecanum wheels with the PID value
```
Comment (tmicro) [inline Team6220_2021/src/main/java/org/firstinspires/ftc/team6220_2021/MasterAutonomous.java:42]: The robot does not have mecanum wheels this season.

File: Team6220_2021/src/main/java/org/firstinspires/ftc/team6220_2021/MasterAutonomous.java [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+            }
+
+            if (Math.abs(targetAngle - angleTraveled) < 1) {
+                driveTank(0.0, 0.0);
+                angleReached = true;
+            }
+        }
+    }
+
+    /*//todo - fix method
```
Comment (tmicro) [inline Team6220_2021/src/main/java/org/firstinspires/ftc/team6220_2021/MasterAutonomous.java:106]: should remove unused code

File: Team6220_2021/src/main/java/org/firstinspires/ftc/team6220_2021/MasterOpMode.java [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+        parameters.angleUnit = BNO055IMU.AngleUnit.DEGREES;
+        parameters.accelUnit = BNO055IMU.AccelUnit.METERS_PERSEC_PERSEC;
+        parameters.calibrationDataFile = "AdafruitIMUCalibration.json"; // see the calibration sample opmode
+        parameters.loggingEnabled = true;
+        parameters.loggingTag = "IMU";
+        imu = hardwareMap.get(BNO055IMU.class, "imu");
+        imu.initialize(parameters);
+    }
+
+    // This method drives mecanum when given an angle drive power and turning power
```
Comment (tmicro) [inline Team6220_2021/src/main/java/org/firstinspires/ftc/team6220_2021/MasterOpMode.java:57]: the robot does not have mecanum wheels this season.

File: Team6220_2021/src/main/java/org/firstinspires/ftc/team6220_2021/ResourceClasses/DriverInput.java [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+
+public class DriverInput {
+    private final Gamepad controller;
+    private final boolean[] buttonStates;  // One state for each button; there are 14 buttons available
+    private final boolean[] lastButtonStates;
+    private final double[] buttonHeldCounts;
+    private double startTime = System.currentTimeMillis();
+
+
+    public DriverInput(Gamepad cont) {
```
Comment (tmicro) [inline Team6220_2021/src/main/java/org/firstinspires/ftc/team6220_2021/ResourceClasses/DriverInput.java:22]: document what this method does.

File: Team6220_2021/src/main/java/org/firstinspires/ftc/team6220_2021/TeleOpCompetition.java [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+import org.firstinspires.ftc.team6220_2021.ResourceClasses.Button;
+
+@TeleOp(name = "TeleOp Competition", group = "TeleOp")
+public class TeleOpCompetition extends MasterTeleOp {
+
+    @Override
+    public void runOpMode() {
+        Initialize();
+
+        DcMotor motorArm;
```
Comment (tmicro) [inline Team6220_2021/src/main/java/org/firstinspires/ftc/team6220_2021/TeleOpCompetition.java:16]: Shouldn't these motors and servos should be instantiated inside the master op mode class?  That way, they can be used for autonomous and teleop.

File: Team6220_2021/src/main/java/org/firstinspires/ftc/team6220_2021/TeleOpCompetition.java [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+        DcMotor motorCarousel;
+        Servo servoGrabber;
+        Servo servoArm;
+
+        motorArm = hardwareMap.dcMotor.get("motorArm");
```
Comment (tmicro) [inline Team6220_2021/src/main/java/org/firstinspires/ftc/team6220_2021/TeleOpCompetition.java:21]: Shouldn't these go inside an initialize() method?

File: Team6220_2021/src/main/java/org/firstinspires/ftc/team6220_2021/TeleOpCompetition.java [hunk 7]
```diff
@@ -0,0 +1,82 @@
...
+        motorCarousel.setMode(DcMotor.RunMode.RUN_WITHOUT_ENCODER);
+
+        waitForStart();
+
+        while (opModeIsActive()) {
+
+            driver1.update();
+            driver2.update();
+
+            driveTankWithJoysticks(1);
```
Comment (tmicro) [inline Team6220_2021/src/main/java/org/firstinspires/ftc/team6220_2021/TeleOpCompetition.java:42]: Did the entire team agree to go with tank drive?  Is that team decision documented in the notebook?

File: Team6220_2021/src/main/java/org/firstinspires/ftc/team6220_2021/MasterAutonomous.java [hunk 8]
```diff
@@ -0,0 +1,141 @@
...
+            position = Constants.IN_PER_AM_TICK * (motorFrontLeft.getCurrentPosition() - motorBackLeft.getCurrentPosition() - motorFrontRight.getCurrentPosition() + motorBackRight.getCurrentPosition()) / 4.0;
+
+            if (position > targetDistance) {
+                driveTank(0.0, 0.0);
+                distanceReached = true;
+            }
+        }
+    }
+
+    public void turnDegrees(double targetAngle) {
```
Comment (tmicro) [inline Team6220_2021/src/main/java/org/firstinspires/ftc/team6220_2021/MasterAutonomous.java:60]: Should document what this method does, and how works.

File: Team6220_2021/src/main/java/org/firstinspires/ftc/team6220_2021/AutonomousCompetition.java [hunk 9]
```diff
@@ -0,0 +1,23 @@
...
+
+import com.qualcomm.robotcore.eventloop.opmode.Autonomous;
+
+import org.firstinspires.ftc.robotcore.external.navigation.AngleUnit;
+import org.firstinspires.ftc.robotcore.external.navigation.AxesOrder;
+import org.firstinspires.ftc.robotcore.external.navigation.AxesReference;
+
+@Autonomous(name = "Autonomous Competition", group = "Autonomous")
+public class AutonomousCompetition extends MasterAutonomous {
+
```
Comment (malavika1) [inline Team6220_2021/src/main/java/org/firstinspires/ftc/team6220_2021/AutonomousCompetition.java:11]: No blank lines after opening curly brace {

File: Team6220_2021/src/main/java/org/firstinspires/ftc/team6220_2021/AutonomousCompetition.java [hunk 10]
```diff
@@ -0,0 +1,23 @@
...
+    @Override
+    public void runOpMode() {
+
+        Initialize();
+        waitForStart();
+
+        telemetry.addData("hi", imu.getAngularOrientation(AxesReference.INTRINSIC, AxesOrder.ZYX, AngleUnit.DEGREES).firstAngle);
```
Comment (malavika1) [inline Team6220_2021/src/main/java/org/firstinspires/ftc/team6220_2021/AutonomousCompetition.java:18]: Do you want to include a more useful string than "hi"? Maybe explain what the value you're printing out is

File: Team6220_2021/src/main/java/org/firstinspires/ftc/team6220_2021/MasterAutonomous.java [hunk 11]
```diff
@@ -1,15 +1,142 @@
...
+            // This adds a value to the PID loop so it can update
+            angleLeft = targetAngle - angleTraveled;
+            translationPID.roll(angleLeft);
+
+            // We drive the mecanum wheels with the PID value
+            if (targetAngle > 0) {
+                driveTank(Math.max((translationPID.getFilteredValue() / 2) * -1, Constants.MINIMUM_TURNING_POWER),
+                        Math.max((translationPID.getFilteredValue() / 5), Constants.MINIMUM_TURNING_POWER));
+            }
+            else {
```
Comment (malavika1) [inline Team6220_2021/src/main/java/org/firstinspires/ftc/team6220_2021/MasterAutonomous.java:94]: The else should be on the previous line. } else {

File: Team6220_2021/src/main/java/org/firstinspires/ftc/team6220_2021/MasterOpMode.java [hunk 12]
```diff
@@ -1,17 +1,68 @@
...
+    public void driveTank(double leftSidePower, double rightSidePower) {
+        motorFrontLeft.setPower(leftSidePower);
+        motorBackLeft.setPower(-leftSidePower);
+        motorFrontRight.setPower(-rightSidePower);
+        motorBackRight.setPower(rightSidePower);
+    }
+
+    // Pauses for time milliseconds
     public void pauseMillis(double time) {
```
Comment (malavika1) [inline Team6220_2021/src/main/java/org/firstinspires/ftc/team6220_2021/MasterOpMode.java:66]: I think pauseMillis was added as a shared method in MasterOpMode. Please use that instead.

File: Team6220_2021/src/main/java/org/firstinspires/ftc/team6220_2021/ResourceClasses/Button.java [hunk 13]
```diff
@@ -0,0 +1,12 @@
+package org.firstinspires.ftc.team6220_2021.ResourceClasses;
+
+/**
+ Identifiers for boolean controls on controller
+ */
+
```
Comment (malavika1) [inline Team6220_2021/src/main/java/org/firstinspires/ftc/team6220_2021/ResourceClasses/Button.java:6]: You don't need a blank line after this comment

File: Team6220_2021/src/main/java/org/firstinspires/ftc/team6220_2021/ResourceClasses/DriverInput.java [hunk 14]
```diff
@@ -0,0 +1,141 @@
...
+ */
+
+public class DriverInput {
+    private final Gamepad controller;
+    private final boolean[] buttonStates;  // One state for each button; there are 14 buttons available
+    private final boolean[] lastButtonStates;
+    private final double[] buttonHeldCounts;
+    private double startTime = System.currentTimeMillis();
+
+
```
Comment (malavika1) [inline Team6220_2021/src/main/java/org/firstinspires/ftc/team6220_2021/ResourceClasses/DriverInput.java:21]: Don't have two blank lines in a row

File: Team6220_2021/src/main/java/org/firstinspires/ftc/team6220_2021/ResourceClasses/DriverInput.java [hunk 15]
```diff
@@ -0,0 +1,141 @@
...
+        // Initialize button arrays
+        buttonStates = new boolean[14];
+        lastButtonStates = new boolean[14];
+        buttonHeldCounts = new double[14];
+        Arrays.fill(buttonStates, false);
+        Arrays.fill(lastButtonStates, false);
+        Arrays.fill(buttonHeldCounts, 0);
+    }
+
+    public void initialize(HardwareMap hMap) {
```
Comment (malavika1) [inline Team6220_2021/src/main/java/org/firstinspires/ftc/team6220_2021/ResourceClasses/DriverInput.java:34]: Do you need this if it's empty?

**Review by malavika1 — CHANGES_REQUESTED** [review]

**Pull request closed without merging**